monty.qmd

```diff
@@ -7,15 +7,94 @@ source("common.R")
+
+The random walk sampler uses a variance-covariance (VCV) matrix to guide its exploration, determining the 'jump' from the current point to the next in a random walk by drawing from a multivariate normal distribution parameterised by this matrix. For our single-parameter model here, we use a 1x1 matrix of variance 2 (`matrix(2)`) as our VCV matrix.
+
+The choice of the VCV matrix is critical for the sampler's efficiency, especially in more complex cases where the tuning of this matrix can significantly affect performance. A well-chosen VCV matrix optimises moving accross the parameter space, making the random walk sampler more effective in exploring the distribution of interest.
+
+```{r}
+sampler <- monty_sampler_random_walk(matrix(2))
+```
+
+We are now ready to sample from our distribution using `monty_sample()` with 2000 samples, starting our MCMC chain at the value 3 and running 4 chains sequentially.
+
+```{r}
+samples <- monty_sample(mixture_model, sampler, 2000, initial = 3, n_chains = 4)
+```
+
+We can visualise our 4 chains.
+```{r}
+matplot(samples$density, type = "l", lty = 1,
+        xlab = "sample", ylab = "log posterior density", col = "#00000055")
+```
+
+We can also check that our samples are correctly representing our distribution:
+```{r}
+hist(samples$pars["l", , ], breaks = 100)
+```
+
+## Connection of `monty` with other software
+
+`monty` includes a simple probabilistic domain-specific language (DSL) that is inspired by languages of the BUGS family such as `stan` and [Statistical Rethinking](https://xcelab.net/rm/). It is designed to make some tasks a bit easier, particularly when defining priors for your model.  We expect that this DSL is not sufficiently advanced to represent most interesting models but it may get more clever and flexible in the future. In particular we do not expect the DSL to be useful in writing likelihood functions for comparison to data; we expect that if your model is simple enough for this you would be better off using `stan` or some similarly flexible system.
+
+*mention SSM, dr jacoby, mcstate, BayesTools
+
+## Going further
+
+We've discussed a little bit about the philosophy of `monty` and built a simple model using an R function with `monty_function`. The example introduces the basics of defining and sampling from a `monty` model.
+
+For more advanced applications, refer to @sec-monty-dsl for constructing models using `monty`'s domain-specific language, which enables greater flexibility for complex structures. To combine multiple models, see @sec-monty-combine, which demonstrates how to compose and integrate existing `monty` models. Lastly, @sec-inference illustrates how to incorporate `odin` models into `monty` workflows, expanding the package’s potential for Bayesian inference with more sophisticated, system-based models. Each section provides detailed examples to guide you in leveraging `monty`’s full capabilities.
```

Comment:
Again, referring to sec-monty-combine here when it doesn't seem to exist

Reply:
Removed ref.